execution_engine/src/shared/newtypes/mod.rs

```diff
@@ -5,14 +5,18 @@ use std::fmt::{self, Display, Formatter};
 use serde::Serialize;
 use uuid::Uuid;
 
+/// A correlation id is a unique representation of an event occuring
+/// in the system as it passes through different components.
```

Comment:
I think this type is only used inside EE as opposed to across different components.
```suggestion
/// A correlation id is a unique identifier which can be used to track the progress of a given
/// execution engine operation.
```